### PR TITLE
exclude date test

### PR DESCRIPTION
**Pull request by anunagula3**

### CODE REVIEW

File: src/test/java/uk/gov/justice/digital/hmpps/vsip/pages/NeedAReviewPage.java [hunk 1]
```diff
@@ -14,21 +14,16 @@ public void isSummaryListOptionDisplayed(String summaryList) {
         methodsService.isElementDisplayed("xpath", "//span[contains(@class, 'govuk-details__summary-text')]");
     }
 
-    public void clickOnView()
-    {
+    public void clickOnView(String excludedate) {
+        String xpath = "//li[@class='govuk-!-margin-bottom-0' and contains(text(), '" + excludedate + "')]";
```
Comment (tpmcgowan) [inline src/test/java/uk/gov/justice/digital/hmpps/vsip/pages/NeedAReviewPage.java:18]: I think we need a different selector for this. `govuk-!-margin-bottom-0` is just a styling class that we use all over the place so can't be relied upon.

Comment (anunagula3) [reply src/test/java/uk/gov/justice/digital/hmpps/vsip/pages/NeedAReviewPage.java]: Done ~ changed the xpath

File: src/test/java/uk/gov/justice/digital/hmpps/vsip/cucumber/steps/ConfirmationSteps.java [hunk 2]
```diff
@@ -24,4 +30,37 @@ public void iSeeABookingReferenceDisplayed() {
...
+        // Parse the original date string
+        LocalDate originalDate = LocalDate.parse(originalDateValue, originalFormatter);
+        // Define the formatter for the desired output format
+        DateTimeFormatter outputFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+        // Format the date to the desired output format
+        String formattedDate = originalDate.format(outputFormatter);
+        //set formatted date to use in further Test steps
+        testContextService.setDateBookingReference(formattedDate);
+        //API call
+        testHelper.addVisitExcludeDateEvent("HEI",formattedDate);
```
Comment (dhirajshettymoj) [inline src/test/java/uk/gov/justice/digital/hmpps/vsip/cucumber/steps/ConfirmationSteps.java:53]: would be ideal to get the prison code dynamically but ok for now.

File: src/test/java/uk/gov/justice/digital/hmpps/vsip/cucumber/steps/ConfirmationSteps.java [hunk 3]
```diff
@@ -24,4 +30,37 @@ public void iSeeABookingReferenceDisplayed() {
...
+
+        // Parse the original date string
+        LocalDate original_DateValue = LocalDate.parse(originalDateValue, DateTimeFormatter.ofPattern("EEEE d MMMM yyyy"));
+
+        // Format the date to the desired output format
+        String truncatedDateValue = originalDate.format(DateTimeFormatter.ofPattern("d MMMM yyyy"));
+
+        // Print the result
+        System.out.println("Original Date value: " + originalDateValue);
+        System.out.println("Truncated Date value: " + truncatedDateValue);
```
Comment (dhirajshettymoj) [inline src/test/java/uk/gov/justice/digital/hmpps/vsip/cucumber/steps/ConfirmationSteps.java:63]: these lines could be moved to a different method.

**Review by dhirajshettymoj — COMMENTED** [review]
2 minor comments added. moving the mentioned lines to a different method will be good as it will allow them to be reused.

File: src/test/java/uk/gov/justice/digital/hmpps/vsip/cucumber/steps/ConfirmationSteps.java (hunk 3 above)
Comment (anunagula3) [reply src/test/java/uk/gov/justice/digital/hmpps/vsip/cucumber/steps/ConfirmationSteps.java:63]: Yes I'm going to do it in a separate PR.

**Review by dhirajshettymoj — APPROVED** [review]
LGTM

**Pull request merged**